spec.md

```diff
@@ -388,13 +388,34 @@ both appear within the same JSON object.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
+### application-properties
+* Type: `Map`
+* Description: A list of "key-value" pairs providing information about the
+  event, which are used by an event-based application. The "key-value" pairs
+  MUST be in a flat structure. The value MUST be a string. There MUST be no
```

Comment:
I think the sentence about "flat structure" is redundant and can be removed since the value MUST be a string implies it.  If not, then I'm not sure what "flat structure" means and it may need to be defined.

Reply:
isn't "MUST be a string" even more restrictive than "flat structure" since it also precludes numeric or boolean values?